src/document/doc-debug.js

```diff
@@ -127,8 +145,33 @@ function printDoc(doc) {
   throw new Error("Unknown doc type " + doc.type);
 }
 
-module.exports = {
-  printDocToDebug(doc) {
-    return printDoc(flattenDoc(doc));
-  },
-};
+let symbolMap;
+let usedStrings;
+
+function printGroupId(id) {
+  if (typeof id !== "symbol") {
+    return String(id);
+  }
+
+  if (id in symbolMap) {
+    return symbolMap[id];
+  }
+
+  const prefix = id.description || "symbol";
+  for (let counter = 0; ; counter++) {
+    const string = prefix + (counter > 0 ? ` #${counter}` : "");
+    if (!usedStrings.has(string)) {
+      symbolMap[id] = string;
+      usedStrings.add(string);
```

Comment:
I don't understand this part, why need two object save the same thing?

Reply:
Symbols can't made round-trippable, so we need to generate unique strings instead.
`symbolMap` is used to look up by a symbol, whether it already has an associated string.
`usedStrings` - to look up by a string, to check whether the given string is already in use for some other symbol.

Reply:
And use a shared variable look unusual to me, I would do

```js
function printDoc(doc) {
  const symbolMap, usedStrings

  function theRealPrintToRecursion() {}

  return theRealPrintToRecursion(doc)
}
```

Reply:
Just realized I can generate `Symbol.for("unique string")` instead of just "unique string". I'm going to fix this.

Reply:
> shared variable look unusual to me

It's fine as long as the function is sync and doesn't reenter itself, we have this in `doc-printer` too: see `groupModeMap`. Probably I should've fixed `lineSuffixBoundary` the same way, by extracting the `lineSuffix` variable to the outer scope instead of adding parameters to `fits`.

> I would do

Agree. Actually, I did that too, but then I decided to minimize the diff to make this PR easier to review. :)

Reply:
Right, I saw `groupModeMap`, don't like it.

Reply:
Okay, I'll wrap these variables in a function scope this PR. And let's do the same with `groupModeMap` and `lineSuffix` in another (Do you want to give it a shot?).

Reply:
All yours, it's late in China, and I have a slow net connection.

Reply:
As you wish. It doesn't have to be today anyway.